web/pdf_attachment_viewer.js

```diff
@@ -120,7 +130,33 @@ var PDFAttachmentViewer = (function PDFAttachmentViewerClosure() {
       }
 
       this._dispatchEvent(attachmentsCount);
-    }
+    },
+
+    /**
+     * Used to append FileAttachment annotations to the sidebar.
+     * @private
+     */
+    _appendAttachment: function PDFAttachmentViewer_appendAttachment(item) {
+      this._renderedCapability.promise.then(function (id, filename, content) {
+        var attachments = this.attachments;
+
+        if (!attachments) {
+          attachments = Object.create(null);
+        } else {
+          for (var name in attachments) {
+            if (id === attachments[name]) {
```

Comment:
This condition doesn't actually make sense, it should obviously be `if (id === name) {` instead.